x-pack/plugins/alerting/server/routes/rules_settings/apis/update/update_query_delay_settings.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IRouter } from '@kbn/core/server';
+import { ILicenseState } from '../../../../lib';
+import { verifyAccessAndContext } from '../../../lib';
+import { AlertingRequestHandlerContext, INTERNAL_BASE_ALERTING_API_PATH } from '../../../../types';
+import { API_PRIVILEGES } from '../../../../../common';
+import {
+  updateQueryDelaySettingsBodySchemaV1,
+  UpdateQueryDelaySettingsRequestBodyV1,
+  UpdateQueryDelaySettingsResponseV1,
+} from '../../../../../common/routes/rules_settings/apis/update';
+import { transformQueryDelaySettingsToResponseV1 } from '../../transforms';
+
+export const updateQueryDelaySettingsRoute = (
+  router: IRouter<AlertingRequestHandlerContext>,
+  licenseState: ILicenseState
+) => {
+  router.post(
+    {
+      path: `${INTERNAL_BASE_ALERTING_API_PATH}/rules/settings/_query_delay`,
+      validate: {
+        body: updateQueryDelaySettingsBodySchemaV1,
+      },
+      options: {
+        tags: [`access:${API_PRIVILEGES.WRITE_QUERY_DELAY_SETTINGS}`],
+      },
+    },
+    router.handleLegacyErrors(
+      verifyAccessAndContext(licenseState, async function (context, req, res) {
+        const rulesSettingsClient = (await context.alerting).getRulesSettingsClient();
+
+        const body: UpdateQueryDelaySettingsRequestBodyV1 = req.body;
+
+        const updatedQueryDelaySettings = await rulesSettingsClient.queryDelay().update(body);
+
+        const response: UpdateQueryDelaySettingsResponseV1 = {
+          body:
+            updatedQueryDelaySettings &&
```

Comment:
when will `updatedQueryDelaySettings`? I see that the update function can throw an error but that wouldn't return as undefined would it?

Reply:
Let me double check, this was the way the flapping api was working and I think left over from the before the changes to split the rules settings into separate objects.

Reply:
Resolved in this commit https://github.com/elastic/kibana/pull/167433/commits/e32a0a8d8d2b7e5d5770f153996463a806002456